### PR TITLE
AG-10431 Update data service to silently handle callback errors

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-10431

Also adds a small delay on the loading overlay to prevent too much flashing

### CODE REVIEW

File: packages/ag-charts-community/src/chart/data/dataService.ts [hunk 1]
```diff
@@ -111,26 +111,25 @@ export class DataService<D extends object> extends Listeners<EventType, EventHan
...
         this.freshRequests.push(id);
 
+        let response;
         try {
-            const response = await this.dataSourceCallback(params);
+            response = await this.dataSourceCallback(params);
             this.debug(`DataService - response | ${performance.now() - start}ms | ${id}`);
+        } catch (error) {
+            this.debug(`DataService - error | ${id}`);
+            // Ignore errors in callback and keep chart alive
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataService.ts:121]: Nit: Maybe a `Logger.errorOnce()` is warranted in this case?

File: packages/ag-charts-community/src/chart/data/dataService.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export class DataService<D extends object> extends Listeners<EventType, EventHan
 
     private createThrottledDispatch(dispatchThrottle: number) {
         return throttle(
-            (id: number, data: D[]) => {
+            (id: number, data: D[] | any) => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataService.ts:91]: Needed?

**Review by alantreadway — APPROVED** [review]
LGTM, merge when ready

File: packages/ag-charts-community/src/chart/data/dataService.ts (hunk 2 above)
Comment (lsjroberts) [reply packages/ag-charts-community/src/chart/data/dataService.ts:91]: The response from the callback is `unknown`, in the future we'll handle this better I presume and only pass through valid data. But for now I added this to let the response always be dispatched.

**Pull request merged**